suite-native/module-trading/src/components/general/SearchInputWithCancel.tsx

```diff
@@ -0,0 +1,49 @@
+import { useRef, useState } from 'react';
+import { TextInput } from 'react-native';
+
+import { Box, HStack, SearchInput, SearchInputProps, TextButton } from '@suite-native/atoms';
+import { Translation, useTranslate } from '@suite-native/intl';
+
+export type SearchInputWithCancelProps = Omit<SearchInputProps, 'placeholder' | 'elevation'>;
+
+const noOp = () => {};
+
+export const SearchInputWithCancel = ({
+    onFocus = noOp,
+    onBlur = noOp,
+    ...props
+}: SearchInputWithCancelProps) => {
+    const { translate } = useTranslate();
+    const [isInputActive, setIsInputActive] = useState(false);
+    const inputRef = useRef<TextInput>(null);
+
+    const handleCancel = () => {
+        inputRef.current?.clear();
+        inputRef.current?.blur();
+    };
+
+    return (
+        <HStack alignItems="center">
+            <Box flex={1}>
```

Comment:
Should be just few lines to add layout animations for showing cancel button, please add it

Reply:
Updated